articles/150_roslaunch.md

```diff
@@ -0,0 +1,364 @@
+author: '[William Woodall](https://github.com/wjwwood)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Context
+
+<div class="alert alert-warning" markdown="1">
+TODO: Fill out some context with respect to ROS 1's `roslaunch` and other similar tools which exist in OS's.
+</div>
+
+## Separation of Concern
+
+The launch system can be considered in parts, separated by concern.
+The coarse breakdown is like so:
+
+- Calling Conventions for Processes and Various Styles of Nodes
+- Reporting System for Event
+- System Description and Static Analysis
+- Execution and Verification of the System Description
+- Testing
+
+The purpose of the following sections is to enumerate what the launch system could interact with or do, but is not the requirements list for the launch system in ROS 2.
+The requirements for the launch system will be enumerated in section below based on what's possible in these sections.
+
+## Calling Conventions
+
+In order for the launch system to execute a described system, it needs to understand how it can achieve the description.
+This is an existing phrase in Computer Science[^calling_convention_wikipedia], but this section is not talking specifically about the compiler defined calling convention, through it is appropriating the term to describe a similar relationship.
+In this case, the phrase "calling conventions" is meant to describe the "interface" or "contract" the launch system has with the entities it is executing and monitoring.
+This contract covers initial execution, activity during runtime, signal handling and behavior of the launch system, and shutdown.
+
+### Operating System Processes
+
+The most basic version of these entities, and the foundation for the other entities, are operating system processes.
+
+#### Execution
+
+For these, the launch system needs to know how to execute them, and to do that it needs:
+
+- name of the executable (just the name, relative path, or absolute path)
+- environment variables (`PATH`, `LD_LIBRARY_PATH`, `DL_PRELOAD`, etc...)
+- command line arguments
+- working directory (directory from which to execute the process)
+- launch prefix (used to inject things like `gdb`, `valgrind`, etc...)
+
+<div class="alert alert-warning" markdown="1">
+RFC:
+
+Missing from this list is the user which should be used to execute the process.
+It's possible that it would be necessary or at least useful to change the user based on the launch description.
+However, it can always be done in a user written script and supporting it in our Python implementation in a portable way looks to be difficult.
```

Comment:
Adding a user name would also raise the complex question of how to authenticate. 

I would avoid that here, unless ROS 2 somehow evolves a system-wide security design which depends on it.

Reply:
I tend to agree, but I do think it would be a useful feature at some point. However, I'm concerned it would make launch files less reusable, e.g. you don't want a driver to try to run as "camera_user" when that might be specific to the machine it was originally developed on. In this way, it would need to be used more cautiously and at integration time, like the remote machine functionality.